etl/src/workers/pool.rs

```diff
@@ -6,41 +6,29 @@ use std::sync::Arc;
 use tokio::sync::{Mutex, Notify};
 use tracing::{debug, warn};
 
-use crate::concurrency::future::ReactiveFutureCallback;
 use crate::destination::base::Destination;
-use crate::error::EtlError;
-use crate::error::{ErrorKind, EtlResult};
-use crate::etl_error;
+use crate::error::EtlResult;
 use crate::state::store::base::StateStore;
 use crate::workers::base::{Worker, WorkerHandle};
 use crate::workers::table_sync::{TableSyncWorker, TableSyncWorkerHandle, TableSyncWorkerState};
 
-#[derive(Debug)]
-pub enum TableSyncWorkerInactiveReason {
-    Completed,
-    Errored(String),
-}
-
 #[derive(Debug)]
 pub struct TableSyncWorkerPoolInner {
     /// The table sync workers that are currently active.
     active: HashMap<TableId, TableSyncWorkerHandle>,
-    /// The table sync workers that are inactive, meaning that they are completed or errored.
-    ///
-    /// Having the state of finished workers gives us the power to reschedule failed table sync
-    /// workers very cheaply since the state can be fed into a new table worker future as if it was
-    /// read initially from the state store.
-    // TODO: make it a bounded history to guard memory usage.
-    inactive: HashMap<TableId, Vec<(TableSyncWorkerInactiveReason, TableSyncWorkerHandle)>>,
-    waiting: Option<Arc<Notify>>,
+    /// The table sync workers that are finished, meaning that either they completed or errored.
+    finished: HashMap<TableId, Vec<TableSyncWorkerHandle>>,
```

Comment:
Simplified the interface because we don’t need to capture non-forwarded errors, as they are already forwarded in the `ReactiveFuture` and so accessible via the `wait()` on the handle.